### ISSUE TITLE
Random crash on loading Bundle infos

### ISSUE THREAD

**Issue by mittsh** [algolia/algoliasearch-client-swift#139]
Hey! Sorry I'm reporting a very strange and unique crash, that I think is actually more likely to be related to the system than this Swift client.

A user reported that to us, and we haven't been able to reproduce ourselves.

Context:
- OS: OS X 10.11.6 (El Capitan)
- Library version: 4.0 or 4.0.1
- Swift 3 called from ObjC
- Framework installed via Cocoapods

Here's the shorten crash stack:

```
Date/Time: 2016-09-28 20:42:15.572 -0400
OS Version: Mac OS X 10.11.6 (15G1004)
Report Version: 11
Anonymous UUID: 078D5877-F1F3-241B-3ECD-162E6A6744AD

Time Awake Since Boot: 430 seconds

System Integrity Protection: enabled

Crashed Thread: 0 Dispatch queue: com.apple.main-thread

Exception Type: EXC_BAD_INSTRUCTION (SIGILL)
Exception Codes: 0x0000000000000001, 0x0000000000000000
Exception Note: EXC_CORPSE_NOTIFY

Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0 org.cocoapods.AlgoliaSearch 0x00000001016d6547 _TFC13AlgoliaSearch6ClientcfT5appIDSS6apiKeySS_S0_ + 6311
1 org.cocoapods.AlgoliaSearch 0x00000001016d6636 _TToFC13AlgoliaSearch6ClientcfT5appIDSS6apiKeySS_S0_ + 102
2 com.luckymarmot.Paw 0x0000000101100aa3 0x101034000 + 838307
3 com.luckymarmot.Paw 0x00000001010ec78f 0x101034000 + 755599
4 com.apple.CoreFoundation 0x00007fff9912ebbc __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 12
…
```

So the symbolicated call stack where the crash happens is:

```
@objc Client.init(appID : String, apiKey : String) -> Client (in AlgoliaSearch) (Client.swift:0)
Client.init(appID : String, apiKey : String) -> Client (in AlgoliaSearch) (Client.swift:197)
```

This leads us to [this line](https://github.com/algolia/algoliasearch-client-swift/blob/master/Source/Client.swift#L197):

```
let version = Bundle(for: type(of: self)).infoDictionary!["CFBundleShortVersionString"] as! String
```

Which is just an access to the `Bundle` (…`NSBundle`) of the framework, and get the version string. Seems like a very innocent call.

Again, that's a random crash and only 1 user reported that to us.

Any idea? Suggestions? Anyone else had issues with just loading the Bundle infos? Thanks!

**Comment by clement-leprovost** [comment]
Hello @mittsh! And thanks for reporting this issue.

It is indeed very strange. I could not reproduce it myself, neither on macOS nor iOS, neither from Objective-C nor Swift, neither in an application nor in unit tests... I have the exact same version of macOS as the reporter (10.11.6).

Do you know whether this crash happens systematically or at random? In a "real" application or inside unit tests?

A very simple workaround would be to hard-code the version number inside a string instead of reading it dynamically from the bundle. It is less elegant and probably more error prone for maintainers, but it would be OK. What worries me is that this crash might hide some deeper issue, in which case addressing just the version number problem won't really help.

Please let me know whether you have more information.

**Comment by mittsh** [comment]
@clement-leprovost Sorry for the super late answer on this! To be honest, I'm tempted to close this issue as I don't have much informations about this.

We haven't been able to reproduce this bug ourselves, and clearly hardcoding the framework version into the code isn't a good solution either. I clearly don't recommend this for this library.

We've patched our app to not make this library be `init` at app startup, so at least it won't crash the whole app if this happens again (might crash when users try to use the search features, which is already a little better…).

Sorry I took your time for a not so helpful issue. Maybe if someone else has this issue, we can continue to use this thread to gather infos? I'm also tempted to search of file a radar for this.…
